spring-cloud-function-gateway/src/main/java/org/springframework/cloud/function/gateway/ProxyExchange.java

```diff
@@ -0,0 +1,611 @@
+        return this;
+    }
+
+    /**
+     * Additional headers, or overrides of the incoming ones, to be used in the downstream
+     * call.
+     * 
+     * @param headers the http headers to use in the downstream call
+     * @return this for convenience
+     */
+    public ProxyExchange<T> headers(HttpHeaders headers) {
+        this.headers.putAll(headers);
+        return this;
+    }
+
+    /**
+     * Sets the names of sensitive headers that are not passed downstream to the backend
+     * service.
+     * 
+     * @param names the names of sensitive headers
+     * @return this for convenience
+     */
+    public ProxyExchange<T> sensitive(String... names) {
+        if (this.sensitive == null) {
+            this.sensitive = new HashSet<>();
+        }
+        for (String name : names) {
+            this.sensitive.add(name.toLowerCase());
+        }
+        return this;
+    }
+
+    /**
+     * Sets the uri for the backend call when triggered by the HTTP methods.
+     * 
+     * @param uri the backend uri to send the request to
+     * @return this for convenience
+     */
+    public ProxyExchange<T> uri(String uri) {
+        try {
+            this.uri = new URI(uri);
+        }
+        catch (URISyntaxException e) {
+            throw new IllegalStateException("Cannot create URI", e);
+        }
+        return this;
+    }
+
+    public String path() {
+        return (String) this.webRequest.getAttribute(
+                HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE,
+                WebRequest.SCOPE_REQUEST);
+    }
+
+    public String path(String prefix) {
+        String path = path();
+        if (!path.startsWith(prefix)) {
+            throw new IllegalArgumentException(
+                    "Path does not start with prefix (" + prefix + "): " + path);
+        }
+        return path.substring(prefix.length());
+    }
+
+    public void forward(String path) {
+        HttpServletRequest request = this.webRequest
+                .getNativeRequest(HttpServletRequest.class);
+        HttpServletResponse response = this.webRequest
+                .getNativeResponse(HttpServletResponse.class);
+        try {
+            request.getRequestDispatcher(path).forward(
+                    new BodyForwardingHttpServletRequest(request, response), response);
+        }
+        catch (Exception e) {
+            throw new IllegalStateException("Cannot forward request", e);
+        }
+    }
+
+    public ResponseEntity<T> get() {
+        RequestEntity<?> requestEntity = headers((BodyBuilder) RequestEntity.get(uri))
+                .build();
+        return exchange(requestEntity);
+    }
+
+    public <S> ResponseEntity<S> get(
+            Function<ResponseEntity<T>, ResponseEntity<S>> converter) {
+        RequestEntity<?> requestEntity = headers((BodyBuilder) RequestEntity.get(uri))
+                .build();
+        return converter.apply(exchange(requestEntity));
```

Comment:
this method body could just be `return converter.apply(get());`, right? (same for other methods here that have a simple + `Converter` variant)